cpp/src/arrow/util/value_parsing.h

```diff
@@ -40,6 +40,8 @@
 
 namespace arrow {
 
+#define FASTFLOAT_ALLOWS_LEADING_PLUS 1
+
```

Comment:
Can we move this to `.cc` (`cpp/src/arrow/util/value_parsing.cc`)?
I think that we don't need to export this to other files.